### PR TITLE
Enable fixes/assits to edit other files

### PR DESCRIPTION
**Pull request by matthiaslarisch**

### CODE REVIEW

File: packages/custom_lint_core/lib/src/change_reporter.dart [hunk 1]
```diff
@@ -76,9 +76,13 @@ abstract class ChangeBuilder {
...
   /// The builder passed to the [buildFileEdit] function has no special support
   /// for any particular kind of file.
+  ///
+  /// Use the [customPath] if the collection of edits should be written to another
+  /// file.
   void addGenericFileEdit(
-    void Function(analyzer_plugin.FileEditBuilder builder) buildFileEdit,
-  );
+    void Function(analyzer_plugin.FileEditBuilder builder) buildFileEdit, {
+    String? customPath,
```
Comment (rrousselGit) [inline packages/custom_lint_core/lib/src/change_reporter.dart:84]: We'd need this on all `addXFileEdit`, such as addDartFileEdit

**Comment by rrousselGit** [comment]
Could you add tests for this? 

Also please sign the CLA :pray: (cf above)

File: packages/custom_lint_core/lib/src/change_reporter.dart (hunk 1 above)
Comment (matthiaslarisch) [reply packages/custom_lint_core/lib/src/change_reporter.dart:88]: Done. `customPath `parameter added to all `addXFileEdit` methods

**Comment by matthiaslarisch** [comment]
I've added the `customPath `parameter to all `addXFileEdit` methods but I'm not sure what kind of test do you want.

**Comment by rrousselGit** [comment]
We'd need automated tests invoking those methods with the new parameters.

**Comment by matthiaslarisch** [comment]
I have checked the tests which are available and the changes of this pull request breaks none of them.
As the changes of this pull request will only be effective when the customPath is explicitly set, no one of the current users will be effected, so I think this change is harmless.
I also looked into the current tests but I'm not sure how I can implement sufficient tests.
Because of the mentioned reasons I will not provide any tests.

**Comment by laurentschall** [comment]
I'll try to find the time to add tests.

**Comment by laurentschall** [comment]
This pull request was included in the [pull request 199](https://github.com/invertase/dart_custom_lint/pull/199) which are the same codes changes but with matching tests.

**Pull request closed without merging**